keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -91,6 +93,31 @@ class ApplicationService(
     repository.deletePin(config, targetEnvironment)
   }
 
+  fun markAsVetoedIn(application: String, veto: EnvironmentArtifactVeto, force: Boolean) {
```

Comment:
Thank you for adding these here. I _really_ feel service objects are a good pattern to follow because not only do they make the controllers lean (as they should be), but makes it much easier to write tests without the whole Spring paraphernalia! ❤️

Reply:
Yes! I love having the service objects!